ADAL/src/telemetry/ADTelemetryAPIEvent.m

```diff
@@ -30,84 +33,152 @@ - (void)setResultStatus:(ADAuthenticationResultStatus)status
 }
 
 - (void)setIsExtendedLifeTimeToken:(NSString*)isExtendedLifeToken
 {
-    [self setProperty:@"is_extended_life_time_token" value:isExtendedLifeToken];
+    [self setProperty:AD_TELEMETRY_IS_EXTENED_LIFE_TIME_TOKEN value:isExtendedLifeToken];
 }
 
 - (void)setErrorCode:(NSString*)errorCode
 {
-    [self setProperty:@"error_code" value:errorCode];
+    [self setProperty:AD_TELEMETRY_ERROR_CODE value:errorCode];
 }
 
 - (void)setProtocolCode:(NSString*)protocolCode
 {
-    [self setProperty:@"protocol_code" value:protocolCode];
+    [self setProperty:AD_TELEMETRY_PROTOCOL_CODE value:protocolCode];
 }
 
 - (void)setErrorDescription:(NSString*)errorDescription
 {
-    [self setProperty:@"error_description" value:errorDescription];
+    [self setProperty:AD_TELEMETRY_ERROR_DESCRIPTION value:errorDescription];
 }
 
 - (void)setErrorDomain:(NSString*)errorDomain
 {
-    [self setProperty:@"error_domain" value:errorDomain];
+    [self setProperty:AD_TELEMETRY_ERROR_DOMAIN value:errorDomain];
 }
 
 - (void)setAuthorityValidationStatus:(NSString*)status
 {
-    [self setProperty:@"validation_status" value:status];
+    [self setProperty:AD_TELEMETRY_AUTHORITY_VALIDATION_STATUS value:status];
 }
 
 - (void)setAuthority:(NSString*)authority
 {
-    [self setProperty:@"authority" value:authority];
+    [self setProperty:AD_TELEMETRY_AUTHORITY value:authority];
+    
+    // set authority type
+    NSString* authorityType = AD_TELEMETRY_AUTHORITY_AAD;
+    if ([ADHelpers isADFSInstance:authority])
+    {
+        authorityType = AD_TELEMETRY_AUTHORITY_ADFS;
+    }
+    [self setProperty:AD_TELEMETRY_AUTHORITY_TYPE value:authorityType];
 }
 
 - (void)setGrantType:(NSString*)grantType
 {
-    [self setProperty:@"grant_type" value:grantType];
+    [self setProperty:AD_TELEMETRY_GRANT_TYPE value:grantType];
 }
 
 - (void)setAPIStatus:(NSString*)status
 {
-    [self setProperty:@"api_status" value:status];
+    [self setProperty:AD_TELEMETRY_API_STATUS value:status];
 }
 
 - (void)setApiId:(NSString*)apiId
 {
-    [self setProperty:@"api_id" value:apiId];
+    [self setProperty:AD_TELEMETRY_API_ID value:apiId];
+}
+
+- (void)setPromptBehavior:(ADPromptBehavior)promptBehavior
+{
+    NSString* promptBehaviorString = nil;
+    switch (promptBehavior) {
+        case AD_PROMPT_AUTO:
+            promptBehaviorString = @"AD_PROMPT_AUTO";
+            break;
+        case AD_PROMPT_ALWAYS:
+            promptBehaviorString = @"AD_PROMPT_ALWAYS";
+            break;
+        case AD_PROMPT_REFRESH_SESSION:
+            promptBehaviorString = @"AD_PROMPT_REFRESH_SESSION";
+            break;
+        case AD_FORCE_PROMPT:
+            promptBehaviorString = @"AD_FORCE_PROMPT";
+            break;
+        default:
+            promptBehaviorString = AD_TELEMETRY_UNKNOWN;
+    }
+    
+    [self setProperty:AD_TELEMETRY_PROMPT_BEHAVIOR value:promptBehaviorString];
+}
+
+- (void)processEvent:(NSMutableDictionary*)eventToBeDispatched
```

Comment:
It's extremely unclear to me what exactly this method is (and its superclass implementation) are trying to accomplish. Please add some comments explaining their purpose and the expected result.